src/shell.js

```diff
@@ -224,8 +255,58 @@ else if (ENVIRONMENT_IS_WEB || ENVIRONMENT_IS_WORKER) {
   }
 }
 else {
-  // Unreachable because SHELL is dependant on the others
-  throw 'Unknown runtime environment. Where are we?';
+  // Unreachable because SHELL is dependent on the others
+  throw new Error('Unknown runtime environment. Where are we?');
+}
+
+// Converts a string of base64 into a byte array.
+// Throws error on invalid input.
+function base64ToBytes(s) {
+  try {
+    var decoded = atob(s);
+    var bytes = new Uint8Array(decoded.length);
+    for (i = 0 ; i < decoded.length ; ++i) {
+      bytes[i] = decoded.charCodeAt(i);
+    }
+    return bytes;
+  } catch (_) {
+    throw new Error('Converting base64 string to bytes failed.');
+  }
+}
+
+// Converts a byte array into a string.
+// Returns invalid output on invalid input.
+function bytesToString(bytes) {
```

Comment:
Ah, yeah, looks like that does exactly the same thing. Is it safe to assume that functions from preamble.js are accessible from shell.js?

If so, I can just merge the `TextDecoder` portion of `bytesToString` into `intArrayToString` and delete `bytesToString`. Should `base64ToBytes` and/or `tryParseAsDataURI` be moved as well?